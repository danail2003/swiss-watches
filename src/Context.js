import React from 'react';

const Context = React.createContext({
    user: null,
    logIn: () => {},
    logOut: () => {}
});

export default Context;