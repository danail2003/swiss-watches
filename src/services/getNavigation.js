const getNavigation = () => {
    const links = [
        {
            title: 'Home',
            link: '/'
        },
        {
            title: 'About us',
            link: '/about'
        },
        {
            title: 'Register',
            link: '/register'
        },
        {
            title: 'Login',
            link: '/login'
        },
        {
            title: 'Contact us',
            link: '/contact'
        }
    ]

    return links
}

export default getNavigation;