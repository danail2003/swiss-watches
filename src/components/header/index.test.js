import React from 'react';
import { render } from '@testing-library/react';
import Header from './index';
import Context from '../../Context';
import { Route, BrowserRouter } from 'react-router-dom';
import getNavigation from '../../getNavigation';

describe('Header', () => {
    it('should return correct value after authenticate', () => {
        const links = getNavigation({
            loggedIn: true,
        });

        const userLinks = [
            {
                title: 'Home',
                link: '/'
            },
            {
                title: 'Account',
                link: '/account'
            },
            {
                title: 'Add',
                link: '/create'
            },
            {
                title: 'Watches',
                link: '/watches'
            },
            {
                title: 'About us',
                link: '/about'
            },
            {
                title: 'Contact us',
                link: '/contact'
            }
        ];

        expect(links).toStrictEqual(userLinks);
    });

    it('should return correct value if is not authenticated', () => {
        const links = getNavigation({
            loggedIn: false
        });

        const guestLinks = [
            {
                title: 'Home',
                link: '/'
            },
            {
                title: 'Watches',
                link: '/watches'
            },
            {
                title: 'Register',
                link: '/register'
            },
            {
                title: 'Login',
                link: '/login'
            },
            {
                title: 'About us',
                link: '/about'
            },
            {
                title: 'Contact us',
                link: '/contact'
            }
        ];

        expect(links).toStrictEqual(guestLinks);
    });

    it('should render authenticated routes', () => {
        const component = render(
            <BrowserRouter>
                <Route>
                    <Context.Provider value={{
                        user: {
                            loggedIn: true
                        }
                    }}>
                        <Header />
                    </Context.Provider>
                </Route>
            </BrowserRouter>
        );

        expect(component).toMatchSnapshot();
    });

    it('should render non-authenticated routes', () => {
        const component = render(
            <BrowserRouter>
                <Route>
                    <Context.Provider value={{
                        user: {
                            loggedIn: false
                        }
                    }}>
                        <Header />
                    </Context.Provider>
                </Route>
            </BrowserRouter>
        );

        expect(component).toMatchSnapshot();
    });
});