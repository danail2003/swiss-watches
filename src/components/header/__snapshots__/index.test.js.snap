// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render authenticated routes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="heading"
      >
        <div />
        <div>
          <div
            class="heading-logo"
          >
            <h3>
              Swiss
              <img
                alt="logo"
                class="logo"
                src="rolex_logo.png"
              />
              Watches
            </h3>
            <nav
              class="nav"
            >
              <ul>
                <li>
                  <a
                    class="link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/account"
                  >
                    Account
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/create"
                  >
                    Add
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/watches"
                  >
                    Watches
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/about"
                  >
                    About us
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/contact"
                  >
                    Contact us
                  </a>
                </li>
                <input
                  placeholder="Search..."
                />
                <i
                  class="fas fa-search"
                />
              </ul>
            </nav>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="heading"
    >
      <div />
      <div>
        <div
          class="heading-logo"
        >
          <h3>
            Swiss
            <img
              alt="logo"
              class="logo"
              src="rolex_logo.png"
            />
            Watches
          </h3>
          <nav
            class="nav"
          >
            <ul>
              <li>
                <a
                  class="link"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/account"
                >
                  Account
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/create"
                >
                  Add
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/watches"
                >
                  Watches
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/about"
                >
                  About us
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/contact"
                >
                  Contact us
                </a>
              </li>
              <input
                placeholder="Search..."
              />
              <i
                class="fas fa-search"
              />
            </ul>
          </nav>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header should render non-authenticated routes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="heading"
      >
        <div />
        <div>
          <div
            class="heading-logo"
          >
            <h3>
              Swiss
              <img
                alt="logo"
                class="logo"
                src="rolex_logo.png"
              />
              Watches
            </h3>
            <nav
              class="nav"
            >
              <ul>
                <li>
                  <a
                    class="link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/watches"
                  >
                    Watches
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/register"
                  >
                    Register
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/login"
                  >
                    Login
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/about"
                  >
                    About us
                  </a>
                </li>
                <li>
                  <a
                    class="link"
                    href="/contact"
                  >
                    Contact us
                  </a>
                </li>
                <input
                  placeholder="Search..."
                />
                <i
                  class="fas fa-search"
                />
              </ul>
            </nav>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="heading"
    >
      <div />
      <div>
        <div
          class="heading-logo"
        >
          <h3>
            Swiss
            <img
              alt="logo"
              class="logo"
              src="rolex_logo.png"
            />
            Watches
          </h3>
          <nav
            class="nav"
          >
            <ul>
              <li>
                <a
                  class="link"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/watches"
                >
                  Watches
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/register"
                >
                  Register
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/login"
                >
                  Login
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/about"
                >
                  About us
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/contact"
                >
                  Contact us
                </a>
              </li>
              <input
                placeholder="Search..."
              />
              <i
                class="fas fa-search"
              />
            </ul>
          </nav>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
